### ISSUE TITLE
OpenSesame experiment fails to start in OS X 10.7.5 and 10.8.5 with symbol not found: ___strlcpy_chk

### ISSUE THREAD

**Issue by nortcele** [smathot/OpenSesame#490]
With OpenSesame 3.1.4.
After having solved issue #489 and updated the MacBook to 10.8.5, I'm still having and experiment failling to start. This time with a `___strlcpy_chk`symbol not found error.
The experiment start properly on macOS 10.12.1.

Error message
```
Stopped

The experiment did not finish normally for the following reason:

Unexpected error
Details

item-stack: ``
exception message: dlopen(/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/site-packages/pygame/base.so, 2): Symbol not found: ___strlcpy_chk Referenced from: /Applications/OpenSesame.app/Contents/Resources/lib/python2.7/site-packages/pygame/../../..//libSDL-1.2.0.dylib Expected in: /usr/lib/libSystem.B.dylib in /Applications/OpenSesame.app/Contents/Resources/lib/python2.7/site-packages/pygame/../../..//libSDL-1.2.0.dylib
time: Thu Nov 24 09:35:47 2016
exception type: ImportError
Traceback (also in debug window)

  File "/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/site-packages/libqtopensesame/runners/inprocess_runner.py", line 38, in execute
    self.experiment.run()
  File "/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/site-packages/libopensesame/experiment.py", line 399, in run
    self.init_display()
  File "/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/site-packages/libopensesame/experiment.py", line 749, in init_display
    canvas.init_display(self)
  File "/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/site-packages/openexp/canvas.py", line 64, in init_display
    cls = backend.get_backend_class(experiment, u'canvas')
  File "/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/site-packages/openexp/backend.py", line 155, in get_backend_class
    return getattr(get_backend_mod(experiment, _type), name)
  File "/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/site-packages/openexp/backend.py", line 133, in get_backend_mod
    return __import__('openexp._%s.%s' % (_type, name), fromlist=['dummy'])
  File "/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/site-packages/openexp/_canvas/xpyriment.py", line 21, in <module>
    from openexp._canvas import canvas
  File "/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/site-packages/openexp/_canvas/canvas.py", line 23, in <module>
    import pygame
  File "/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/site-packages/pygame/__init__.py", line 133, in <module>
    from pygame.base import *
ImportError: dlopen(/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/site-packages/pygame/base.so, 2): Symbol not found: ___strlcpy_chk
  Referenced from: /Applications/OpenSesame.app/Contents/Resources/lib/python2.7/site-packages/pygame/../../..//libSDL-1.2.0.dylib
  Expected in: /usr/lib/libSystem.B.dylib
 in /Applications/OpenSesame.app/Contents/Resources/lib/python2.7/site-packages/pygame/../../..//libSDL-1.2.0.dylib
```

**Comment by nortcele** [comment]
The experiment is set to use the expyriment backend. When using the menu "Tools > System Information" I can see that the expyriment module is not available. Is it normal?

```
System

System: Darwin
Architecture: 64bit
Modules and packages

OpenSesame 3.1.4
Python 2.7.12 |Continuum Analytics, Inc.| (default, Jul 2 2016, 17:43:17) [GCC 4.2.1 (Based on Apple Inc. build 5658) (LLVM build 2336.11.00)]
QProgEdit 4.0.6
datamatrix 0.3.1
qdatamatrix 0.1.9
pseudorandom 0.2.2
fileinspector 1.0.2
QNotifications 1.1.1
QOpenScienceFramework 1.1.7
opencv [not available]
expyriment [not available]
IPython 5.0.0
numpy 1.11.1
scipy 0.17.1
PIL/ PILLOW 1.1.7
psychopy 1.82.01
pygame [not available]
pygaze 0.6.0a21
pyglet 1.2.4
PyQt 4.11.4
serial 2.7
markdown 2.6.6
bidi 0.4.0
yaml 3.11
```

**Comment by dschreij** [comment]
I have built the last version of OpenSesame on Mac OS X El Capitan (10.11) and it is really difficult to keep the app backward compatible with older macs. It appears as if something changed in the `/usr/lib/libSystem.B.dylib` on newer macs, causing Pygame (or SDL on which it depends) to not recognise it. This problem is not easy to solve; It appears to be deeply rooted in Pygame. The only way to fix this is to create the OpenSesame app on a very old OS X version on which we would minimally want it to run. Apps created on older OSX's usually tend to run fine on the newer versions, but I don't have an old OS X installation at hand to do so.

The only possible solution I can think of is that you install OpenSesame from source on such older platforms. For this you need to install the SDL libraries using [homebrew](http://brew.sh/) and then install OpenSesame [using Miniconda/Anaconda or pip](http://osdoc.cogsci.nl/3.1/download/#all-platforms-pip).

**Comment by smathot** [comment]
@dschreij, I figured that this was a nasty one, related to packaging. I suggest that we close this issue. Not because it's not a problem, but because we cannot realistically address it. Do you agree?

@nortcele You mention that you solved #489. Can you say a bit more about that? Do you think there's an actual bug in there that we can fix, or that's a packaging-related issue like this one?

**Comment by nortcele** [comment]
#489 is some sort of similar. In 10.8 some library where moved. CoreGraphics is such a library. It was allready there in 10.7 but in another place. So code linked for later versions can't find the library in 10.7. A symbolic link solve the problem in 10.7.

If this issue is not solved there is no need to solve #489 as OpenSesame is later blocked with this issue.

**Comment by dschreij** [comment]
Yes, it is not realistic that we can solve this without investing a considerable amount of time, which I don't think is worth it. 

It's really a shame, because what Anaconda does well is safeharbouring the backward compatibility of their packages with older OS X versions (by building all their packages on CIs that run on an older version of OSX). Sadly, SDL is not available on anaconda so it is the only package where I copied the homebrew version from my own (newer) system, and that is exactly which causes the problem. A solution would be to build SDL ourselves using the conda CI's, but I'm having enough trouble building SDL2 as it is, so it's not something I'd like to spend time on. 
This is yet another reason in my opinion to move away from pygame eventually ;)
